std/linked_list.zig

```diff
@@ -0,0 +1,271 @@
+const debug = @import("debug.zig");
+const assert = debug.assert;
+const mem = @import("mem.zig");
+const Allocator = mem.Allocator;
+
+/// Generic doubly linked list.
+pub fn LinkedList(comptime T: type) -> type {
+    struct {
+        const List = this;
+
+        /// Node inside the linked list wrapping the actual data.
+        const Node = struct {
```

Comment:
Where would the compile error be?

here?

```
    var one   = %%list.createNode(1);
```

error: `one` has type `Node` which is private
note: declared here